### PR TITLE
[T132352] Add filters to interface, minor styling and semantic changes

### PR DESCRIPTION
**Pull request by MusikAnimal**

### CODE REVIEW

File: public_html/templates/base.html [hunk 1]
```diff
@@ -35,56 +35,79 @@
...
-				{% if flash.info|default(false) %}
-					<div class="alert alert-info alert-dismissable">
-						<button type="button" class="close" data-dismiss="alert" aria-hidden="true">&times;</button>
-						{{ flash.info }}
-					</div>
-				{% endif %}
-			{% endif %}
-		{% endblock flashes %}
+			{% endblock flashes %}
+		</section>
```
Comment (MusikAnimal) [inline public_html/templates/base.html]: This gives the banners under it's own `<section>`

File: public_html/templates/base.html [hunk 2]
```diff
@@ -35,56 +35,79 @@
...
+
+	<section class="records">
+		{% block rowheader %}
+			<div class="header-div container-fluid">
+				<div class="header-col col-sm-3 text-center">Page</div>
+				<div class="header-col col-sm-1 text-center">Diff</div>
+				<div class="header-col col-sm-2 text-center">Editor</div>
+				<div class="header-col col-sm-3 text-center">Wikiprojects</div>
+				<div class="header-col col-sm-2 text-center">Review</div>
+			</div>
```
Comment (MusikAnimal) [inline public_html/templates/base.html]: Same thing, just gives the records its own `<section>`

File: public_html/css/index.css [hunk 3]
```diff
@@ -153,11 +154,11 @@
 
 #top-banner {
     text-align: center;
+    margin-bottom: 20px;
```
Comment (MusikAnimal) [inline public_html/css/index.css]: this keeps the header from overlapping into the subsequent section (banners or filters)

File: public_html/css/index.css [hunk 4]
```diff
@@ -153,11 +154,11 @@
...
     /*border: 1px solid deepskyblue;*/
 }
 
 h1 {
-    width: 300px;
```
Comment (MusikAnimal) [inline public_html/css/index.css]: not needed

File: public_html/css/index.css (hunk 3 above)
Comment (kaldari) [inline public_html/css/index.css]: Please put this note in a comment for future reference.

File: src/Controllers/CopyPatrol.php [hunk 5]
```diff
@@ -87,7 +87,34 @@ public function setWikiprojectDao( $wikiprojectDao ) {
...
+			'noaction' => 'All "no action needed" cases'
+		);
+
+		// use given filter if set, or 'all' if filter is 'mine' and they are not logged in
+		if ( isset( $_GET['filter'] ) ) {
+			if ( $_GET['filter'] === 'mine' ) {
+				if ( isset( $currentUser ) ) {
+					$filter = 'mine';
+					$filterUser = $currentUser;
+					$filterTypes['mine'] = 'My reviews'; // add 'My reviews' to filter options
```
Comment (kaldari) [inline src/Controllers/CopyPatrol.php]: It looks like they only get the 'My reviews' option if they've already set the filter to 'mine', which doesn't make sense.

File: src/Controllers/CopyPatrol.php [hunk 6]
```diff
@@ -87,7 +87,34 @@ public function setWikiprojectDao( $wikiprojectDao ) {
...
+
+		// set filter types and descriptions that will be rendered as checkboxes in the view
+		$filterTypes = array(
+			'all' => 'All cases',
+			'open' => 'Open cases',
+			'fixed' => 'All "page fixed" cases',
+			'noaction' => 'All "no action needed" cases'
+		);
+
+		// use given filter if set, or 'all' if filter is 'mine' and they are not logged in
```
Comment (kaldari) [inline src/Controllers/CopyPatrol.php]: I don't see anything about setting it to 'all' in the subsequent code.

**Comment by kaldari** [comment]
+2

File: src/Dao/PlagiabotDao.php [hunk 7]
```diff
@@ -48,12 +48,39 @@ public function __construct( $dsn, $user, $pass, $wiki = 'https://en.wikipedia.o
...
 	 */
-	public function getPlagiarismRecords( $n = 50 ) {
-//		$limit = 'LIMIT ' . $n;
+	public function getPlagiarismRecords( $n = 50, $filter = 'all', $filterUser = NULL ) {
+		$filters = array();
+		$filterSql = '';
+
+		switch ( $filter ) {
+			case 'fixed':
+				array_push( $filters, "status = 'fixed'" );
```
Comment (Niharika29) [inline src/Dao/PlagiabotDao.php]: $filters[] = "status = 'fixed'"; It is the preferred way when doing MW development. Similarly below.

File: src/Controllers/CopyPatrol.php [hunk 8]
```diff
@@ -87,7 +87,37 @@ public function setWikiprojectDao( $wikiprojectDao ) {
...
+			$filterTypes['mine'] = 'My reviews';
+		}
+
+		// use given filter if set, or the default 'open' if filter is 'mine' and user is logged out
+		if ( isset( $_GET['filter'] ) ) {
+			if ( $_GET['filter'] === 'mine' ) {
+				if ( isset( $currentUser ) ) {
+					$filter = 'mine';
+					$filterUser = $currentUser;
+				}
```
Comment (Niharika29) [inline src/Controllers/CopyPatrol.php]: We should handle the else case here. You can have an alert banner which shows up if you set the 'error-illegal' or something set to true in the view.

File: src/Controllers/CopyPatrol.php [hunk 9]
```diff
@@ -87,7 +87,37 @@ public function setWikiprojectDao( $wikiprojectDao ) {
...
+			'fixed' => 'All "page fixed" cases',
+			'noaction' => 'All "no action needed" cases'
+		);
+		// add 'My reviews' to filter options if user is logged in
+		if ( isset( $currentUser ) ) {
+			$filterTypes['mine'] = 'My reviews';
+		}
+
+		// use given filter if set, or the default 'open' if filter is 'mine' and user is logged out
+		if ( isset( $_GET['filter'] ) ) {
```
Comment (Niharika29) [inline src/Controllers/CopyPatrol.php]: We should validate the GET input. Also, check that the input is part of $filterTypes. If it fails the check, show a banner alert (comment below).

File: public_html/templates/base.html [hunk 10]
```diff
@@ -35,56 +35,79 @@
...
-			<div class="header-col col-sm-2 text-center">Review</div>
-		</div>
-	{% endblock %}
-	<div class="record-container">
-		{% block records %}
+
+	<section class="filters container-fluid">
+		<form>
+			<fieldset class="form-group">
+				<legend class="filter-legend">Filter</legend>
```
Comment (Niharika29) [inline public_html/templates/base.html]: You can get rid of the legend. It's pretty clear that it's a filter.

File: public_html/templates/base.html [hunk 11]
```diff
@@ -35,56 +35,79 @@
...
+				{% for type, description in filterTypes %}
+					<span class="radio-inline">
+						<label>
+							<input type="radio" name="filter" value="{{ type }}"
+								{% if filter == type %}checked="checked"{% endif %}
+							/> {{ description }}
+						</label>
+					</span>
+				{% endfor %}
+				<button type="submit" class="btn btn-primary btn-submit-filters">Submit</button>
```
Comment (Niharika29) [inline public_html/templates/base.html]: The button looks a little too big, maybe try btn-sm?

File: public_html/css/index.css [hunk 12]
```diff
@@ -191,3 +192,11 @@ span.cv-hl-in {
...
     background: linear-gradient(to right, #FFC264, #FFF);
 }
+
+.filter-legend {
+    margin-bottom: 10px;
+}
+
+.btn-submit-filters {
+    margin-left: 10px;
+}
```
Comment (Niharika29) [inline public_html/css/index.css]: Add a text-align:center for the fieldset. (class="form-group")

**Comment by Niharika29** [comment]
Looks good. Left some inline comments.

**Pull request merged**